src/main/java/com/puppycrawl/tools/checkstyle/PropertyCacheFile.java

```diff
@@ -220,18 +220,9 @@ public void remove(String checkedFileName) {
      */
     private static String getHashCodeBasedOnObjectContent(Serializable object) {
         try {
-            // im-memory serialization of Configuration
-
             final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
-            ObjectOutputStream oos = null;
-            try {
-                oos = new ObjectOutputStream(outputStream);
-                oos.writeObject(object);
-            }
-            finally {
-                flushAndCloseOutStream(oos);
-            }
-
+            // in-memory serialization of Configuration
+            serialize(object, outputStream);
```

Comment:
you need to move `oos` out of TRY scope and merge TRYs, so `finally` become part of top TRY.